### PR TITLE
fix recursive handling of links

### PR DESCRIPTION
**Pull request by gabemontero**
@bparees PTAL

### CODE REVIEW

**Comment by bparees** [comment]
doesn't look good....
[test]

**Comment by gabemontero** [comment]
There several of these:

```
psql: could not connect to server: Connection refused
	Is the server running on host "172.17.0.2" and accepting
	TCP/IP connections on port 5432?
```

And at the end, one of these:

```
mkdir: cannot create directory '/var/lib/pgsql/data/userdata': Permission denied
```

And there is this in the Dockerfile:
`/usr/libexec/fix-permissions /var/lib/pgsql`

So something is amiss with the finds

**Comment by gabemontero** [comment]
I see what is wrong ... will be submitting update ... my diffs somehow missed the differences in the postgres fix-permission vs. the other ones

**Comment by gabemontero** [comment]
[test]

**Comment by gabemontero** [comment]
ok @bparees the test check is happy again

File: 9.2/root/usr/libexec/fix-permissions [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
+# Fix permissions on the given directory to allow group read/write of 
 # regular files and execute of directories.
-chown -R postgres "$1"
-chgrp -R 0 "$1"
-chmod -R g+rw "$1"
-find "$1" -type d -exec chmod g+x {} +
+find $1 -exec chown postgres {} \;
+find $1 -exec chgrp 0 {} \;
+find $1 -exec chmod g+rw {} \;
+find $1 -type d -exec chmod g+x {} +
```
Comment (bparees) [inline 9.2/root/usr/libexec/fix-permissions]: Quotes should remain for safety (path could contain a space)

Comment (gabemontero) [inline 9.2/root/usr/libexec/fix-permissions]: quotes readded

**Comment by bparees** [comment]
lgtm

**Comment by praiskup** [comment]
This is not nice change :( I miss the reasoning.  Why is 'find -exec' better
than 'chown -R'?

**Comment by bparees** [comment]
@praiskup chmod and chgrp and presumably chown, do not change the ownership of symlinked files when used with -R.

see:
https://github.com/openshift/jenkins/pull/110#discussion_r66736478

**Comment by praiskup** [comment]
Did we found a reasons why coreutils  'chmod' does not support --[no]-dereference option, '-H' or '-L'? Is this really missing feature?

Why we don't use those options where possible?  When we don't have to care about portability,
find && xargs should be used (to not waste forks).

Is this really what we want?  To allow symbolic links and then to fix the destination?

What is the most important message, we should have documentation in commit message.

**Comment by praiskup** [comment]
Note there is also '-exec {} +' variant.  Here is the bug asking for missing '-L' option in chmod:
https://bugzilla.redhat.com/show_bug.cgi?id=1346074

**Pull request merged**